Model/Media.php

```diff
@@ -153,6 +155,19 @@ public function __toString()
         return $this->getName() ?: 'n/a';
     }
 
+    public function __call($method, $arguments)
+    {
+        if ($method == 'setCategory') {
+            $category = current($arguments);
+
+            if (null !== $category && !is_a($category, 'Sonata\ClassificationBundle\Model\CategoryInterface')) {
+                throw new \InvalidArgumentException('Argument 1 should be an instance of Sonata\ClassificationBundle\Model\CategoryInterface or null');
```

Comment:
If you ask me, second solution is the one I like the most.

For me when I see a line super long I just split it in a way that it (more or less) fits well... but I don't have the perfect solution that applies for all the cases.

Also like to follow SF code style rules, but we don't know if they are using the rule of no line breaks on error/exception because they like it or because it's hard to change now.